### ISSUE TITLE
After compiling Umple, pass the results to the base langauge compiler, and compile, pointing any error messages to code embedded in Umple

### ISSUE THREAD

**Issue by Nava2** [umple/umple#95]
Originally reported on Google Code with ID 193
Owned by `redeyedmars`

---

Umple would then appear like a normal compiler, and could be used in the same manner

See also 66

---

Reported by @umple on 2011-06-25 03:10:44
- **Blocked on**: #66

**Comment by Nava2** [comment]
Reported by @umple on 2011-09-28 15:23:51
- **Labels added**: keyPortfolio, ucosp

**Comment by Nava2** [comment]
This has been implements and works in simple cases, but there remain considerable numbers
of bugs, and there wasn't enough testing nor thorough code review.

Also if Java crashes and a stack trace is produced, it would be useful to have a tool
that would map the stack trace line numbers to the original Umple  files

---

Reported by @umple on 2012-06-15 14:37:26
- **Status changed**: `Ongoing`
- **Labels added**: Type-ProjectUG, Diffic-Med
- **Labels removed**: Type-Enhancement, ucosp
- **No longer blocked on**: #22

**Comment by Nava2** [comment]
See also issue 131

---

Reported by @umple on 2012-06-15 15:32:42

**Comment by Nava2** [comment]
The code should be revisited once issue 131 is complete, since more accurate computation
of the originating line number should be possible. Also, new issue 247 (debugger) is
related.

---

Reported by @umple on 2012-10-26 17:29:02
- **Owner removed**

**Comment by Nava2** [comment]
Reported by @umple on 2012-10-27 00:31:56
- **Owner changed**: `andrew.paugh`

**Comment by Nava2** [comment]
Based on Andrew Paugh's work on issue 131, I was able to quickly rewrite the code in
the main program to point out the correct spot in the .ump files instead of the .java
files when the compiler crashes. I also have instrumented the command line option -c
- so that it does the same thing (although there are some bugs remaining in the context
  of complex namespaces and use directives). Work will continue

---

Reported by @umple on 2012-12-11 13:30:11
- **Owner changed**: @umple
- **Labels added**: Priority-VHigh
- **Labels removed**: Priority-Medium
- **cc added:** +`andrew.paugh`

**Comment by Nava2** [comment]
Allthough assigned to me, work on this could continue to be done by a motivated UCOSP
student. Making this work perfectly is very important, as it will facilitate the use
of a debugger directly on Umple code.

---

Reported by @umple on 2012-12-31 14:03:40
- **Labels added**: ucosp
- **cc removed:** -`andrew.paugh`

**Comment by Nava2** [comment]
Reported by @umple on 2013-04-30 13:45:01
- **Owner changed**: `redeyedmars`

**Comment by Nava2** [comment]
Reported by @umple on 2013-07-08 15:08:46
- **Status changed**: `MostlyDone`
- **Labels added**: Type-Enhancement
- **Labels removed**: Type-ProjectUG

**Comment by Nava2** [comment]
Reported by @umple on 2013-09-12 15:12:06
- **Labels removed**: ucosp

**Comment by TimLethbridge** [comment]
#882 is an active issue dealing with this same thing, so this is being closed
